core/tests/test-dependencies/root-files-mapping-generator.ts

```diff
@@ -0,0 +1,723 @@
+      };
+    }
+
+    const objectArgument = decorationNode.getArguments()[0];
+    if (
+      !objectArgument ||
+      !objectArgument.isKind(ts.SyntaxKind.ObjectLiteralExpression)
+    ) {
+      throw new Error(
+        `No object argument found in ${decorationText} on class ` +
+          `${className} in ${sourceFile.getFilePath()}`
+      );
+    }
+
+    const selectorProperty = objectArgument.getProperty('selector');
+    const selector = selectorProperty
+      ? selectorProperty
+          .asKindOrThrow(ts.SyntaxKind.PropertyAssignment)
+          .getInitializerOrThrow()
+          .asKindOrThrow(ts.SyntaxKind.StringLiteral)
+          .getLiteralValue()
+      : undefined;
+    if (type === 'directive') {
+      return {
+        type,
+        className,
+        selector,
+      };
+    }
+
+    const nameProperty = objectArgument.getProperty('name');
+    const name = nameProperty
+      ? nameProperty
+          .asKindOrThrow(ts.SyntaxKind.PropertyAssignment)
+          .getInitializerOrThrow()
+          .asKindOrThrow(ts.SyntaxKind.StringLiteral)
+          .getLiteralValue()
+      : undefined;
+    if (type === 'pipe') {
+      return {
+        type,
+        className,
+        selector: name,
+      };
+    }
+
+    const templateUrlProperty = objectArgument.getProperty('templateUrl');
+    const templateUrl = templateUrlProperty
+      ? templateUrlProperty
+          .asKindOrThrow(ts.SyntaxKind.PropertyAssignment)
+          .getInitializerOrThrow()
+          .asKindOrThrow(ts.SyntaxKind.StringLiteral)
+          .getLiteralValue()
+      : undefined;
+
+    return {
+      type,
+      className,
+      selector,
+      templateFilePath: templateUrl
+        ? resolveModuleRelativeToRoot(templateUrl, sourceFile.getFilePath())
+        : undefined,
+    };
+  });
+};
+
+/**
+ * Checks if a file is a frontend test file.
+ */
+const isFrontendTestFile = (file: string): boolean => {
+  return (
+    file.endsWith('.spec.ts') && !file.includes('puppeteer-acceptance-tests')
+  );
+};
+
+/**
+ * Gets the Angular informations from the given files.
+ */
+const getFileToAngularInformationsFromFiles = (
+  files: string[]
+): Record<string, AngularInformation[]> => {
+  return files.reduce(
+    (acc: Record<string, AngularInformation[]>, file: string) => {
+      if (isFrontendTestFile(file)) {
+        acc[file] = [];
+        return acc;
+      }
+      const sourceFile = project.addSourceFileAtPath(file);
+      const angularInformations =
+        getAngularInformationsFromSourceFile(sourceFile);
+      acc[file] = angularInformations;
+      return acc;
+    },
+    {}
+  );
+};
+
+/**
+ * Checks if the given text contains a specific pipe selector.
+ */
+const isPipeSelectorPresentInText = (
+  text: string,
+  selector: string
+): boolean => {
+  return text.includes('|') && text.includes(selector);
+};
+
+/**
+ * Gets the Angular dependencies from a HTML file.
+ */
+const getAngularDependenciesFromHtmlFile = (
```

Comment:
Is there any other Angular filetype that can be present in HTML apart from directive, pipe, or component?

Reply:
Can't think of any others, do you have any in mind?

Reply:
Honestly I didn't look if there are any other possible Angular filetypes, can you please check it and think whether it makes sense to implement them just to be more future proof?

Reply:
Took a look and couldn't find any other Angular types that can be in a HTML file.

Reply:
Okay!